packages/block-editor/src/components/button-block-appender/index.js

```diff
@@ -60,6 +60,8 @@ function ButtonBlockAppender(
 						onClick={ onToggle }
 						aria-haspopup={ isToggleButton ? 'true' : undefined }
 						aria-expanded={ isToggleButton ? isOpen : undefined }
+						// TODO: Investigate whether this button should be accessible when disabled.
+						// eslint-disable-next-line no-restricted-syntax
```

Comment:
If I did the archeology correctly, it was added here: https://github.com/WordPress/gutenberg/pull/9815

It can now be seen when inserting buttons in the `Buttons` block:

![Screenshot 2024-05-31 at 14 03 26](https://github.com/WordPress/gutenberg/assets/8436925/0cc6824d-4795-42e4-ac09-6badaf51c749)

Judging by the functionality and the initial intent, my expectation is that this `disabled` was left there unintentionally and there's no reason for it to be there in the first place. I expect that the inserter appender will not be visible at all, rather than stay visible and disabled. With that in mind, I don't think it needs to be accessible if it ends up disabled.

Reply:
Thank you so much for investigating! I added a disable reason in d16d2b4.